lib/ObjWriter/objwriter.cpp

```diff
@@ -596,31 +682,42 @@ extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
   EmitLabelDiff(OST, SymbolsBegin, SymbolsEnd);
   OST.EmitLabel(SymbolsBegin);
   {
-    MCSymbol *ProcSegmentBegin = OutContext.createTempSymbol(),
-             *ProcSegmentEnd = OutContext.createTempSymbol();
-    EmitLabelDiff(OST, ProcSegmentBegin, ProcSegmentEnd, 2);
-    OST.EmitLabel(ProcSegmentBegin);
-
-    OST.EmitIntValue(unsigned(SymbolRecordKind::S_GPROC32_ID), 2);
-    // Some bytes of this segment don't seem to be required for basic debugging,
-    // so just fill them with zeroes.
-    OST.EmitFill(12, 0);
-    // This is the important bit that tells the debugger where the function
-    // code is located and what's its size:
-    OST.EmitIntValue(FunctionSize, 4);
-    OST.EmitFill(12, 0);
+    SymRecord RecHeader = {ulittle16_t(sizeof(ulittle16_t) + sizeof(ProcSym) +
+                                       strlen(FunctionName) + 1),
+                           ulittle16_t(SymbolRecordKind::S_GPROC32_ID)};
```

Comment:
I think this looks good to me.
Can you do similar things in other cases like DefRangeRegisterSym, etc?
Ideally I would like it to be function something like this.
```

EmitSymRecord(int size /* the size of the following symbol */, int symbolKind)
{
SymbolRecord SR = { size + sizeof(little16_t), symbolKind }; // include the size of kind.
OST.EmitBytes(StringRef((char *)&SR, sizeof(SR)));
}
```